### PR TITLE
MVP Processors Preview: Support @ProxyEvent

### PR DESCRIPTION
**Pull request by Chris-V**

### CODE REVIEW

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/endpoint/EndPointMethod.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import com.google.common.base.Optional;
 import com.google.common.collect.FluentIterable;
 import com.gwtplatform.dispatch.rest.processors.details.EndPointDetails;
-import com.gwtplatform.dispatch.rest.processors.details.Method;
+import com.gwtplatform.processors.tools.domain.Method;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/endpoint/EndPointMethod.java]: wrong order

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/endpoint/EndPointUtils.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import com.gwtplatform.dispatch.rest.processors.details.Variable;
+import com.gwtplatform.processors.tools.domain.Variable;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/endpoint/EndPointUtils.java]: idem

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/resource/ResourceMethodUtils.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 import com.google.common.base.Function;
 import com.google.common.collect.FluentIterable;
 import com.gwtplatform.dispatch.rest.processors.details.HttpVariable;
-import com.gwtplatform.dispatch.rest.processors.details.Method;
+import com.gwtplatform.processors.tools.domain.Method;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/resource/ResourceMethodUtils.java]: also here

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/subresource/SubResource.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
 import com.google.common.base.Optional;
 import com.google.common.collect.FluentIterable;
 import com.gwtplatform.dispatch.rest.processors.details.EndPointDetails;
-import com.gwtplatform.dispatch.rest.processors.details.Variable;
+import com.gwtplatform.processors.tools.domain.Variable;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/subresource/SubResource.java]: ibid

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/subresource/SubResourceMethod.java [hunk 5]
```diff
@@ -25,7 +25,7 @@
 import com.google.common.collect.FluentIterable;
 import com.gwtplatform.dispatch.rest.processors.details.EndPointDetails;
 import com.gwtplatform.dispatch.rest.processors.details.EndPointDetails.Factory;
-import com.gwtplatform.dispatch.rest.processors.details.Method;
+import com.gwtplatform.processors.tools.domain.Method;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/processors/subresource/SubResourceMethod.java]: again

**Comment by olafleur** [comment]
LGTM. Good job !

**Comment by meriouma** [comment]
LGTM

**Pull request merged**